src/app/reactUtils.ts

```diff
@@ -249,3 +266,41 @@ export const disableContentTabbingHandler = (isEnabled: boolean) => () => {
 export const useDisableContentTabbing = (isEnabled: boolean) => {
   React.useEffect(disableContentTabbingHandler(isEnabled), [isEnabled]);
 };
+
+export type KeyCombinationOptions = Partial<Pick<KeyboardEvent, 'altKey' | 'ctrlKey' | 'key' | 'metaKey' | 'shiftKey'>>;
+
+/**
+ * Attach keydown event listener to the document and check if clicked keys are matching @param options
+ * Value of @param options.key is converted to lowercase before comparing.
+ * @param {KeyCombinationOptions} options
+ * @param {Function} callback
+ */
+export const useKeyCombination = (
+  options: KeyCombinationOptions,
+  callback: () => void,
+  noopHandler?: (activeElement: Element | null) => boolean
+) => {
+  const document = assertDocument();
+
+  const handler = React.useCallback((event: KeyboardEvent) => {
+    if (noopHandler && noopHandler(document.activeElement)) {
+      return;
+    }
+    for (const option in options) {
+      if (
+        (option === 'key' && event.key.toLowerCase() !== options.key!.toLowerCase())
+        || (event[option as keyof KeyCombinationOptions] !== options[option as keyof KeyCombinationOptions])
```

Comment:
i'm not seeing a great way to iterate like this and preserve types. you might try some formatting more like

```
for (const [option, value] of Object.entries(options)) {
```

```suggestion
        || (!(option in event) || event[option] !== value)
```

and maybe some additional thing that checks the types at runtime and does toLowerCase on both if they're strings. 

also consider extracting some of this into a `keybaordEventMatchesCombination` util